broker/src/main/java/io/zeebe/broker/system/partitions/impl/AtomixRecordEntrySupplierImpl.java

```diff
@@ -22,13 +21,12 @@ public AtomixRecordEntrySupplierImpl(final RaftLogReader reader) {
   }
 
   @Override
-  public Optional<Indexed<RaftLogEntry>> getIndexedEntry(final long position) {
-    reader.seekToAsqn(position);
-    if (reader.hasNext()) {
-      final Indexed<RaftLogEntry> entry = reader.next();
-      if (entry.type() == ZeebeEntry.class) {
-        return Optional.of(entry);
-      }
+  public Optional<Indexed<RaftLogEntry>> getPreviousIndexedEntry(final long position) {
+    // Here we are seeking twice. Since this method is only called when taking a snapshot it is ok
+    // to be not very efficient.
+    final long recordIndex = reader.seekToAsqn(position);
+    if (reader.reset(recordIndex - 1) == recordIndex - 1 && reader.hasNext()) {
```

Comment:
Can we extract `reader.reset(recordIndex - 1)` to a variable? It makes it easier when debugging, plus it has side effects, so it's a little clearer to see than in a condition.

Reply:
Nit: Is the `hasNext()` necessary? Since reset returns the next index that `next` would call? I guess it doesn't hurt, but I think it's not necessary?